File: programming-contest/AOJ/ALDS/1_1_D.java
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.BufferedReader;

 
public class Main {
 
    public static final int BIG_NUM  = 2000000000;
 
    public static void main(String[] args) {
        int max = -BIG_NUM, min = BIG_NUM;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
 
        try {
            int N = Integer.parseInt(br.readLine());
 
            for(int loop = 0; loop < N; loop++){
                int tmp = Integer.parseInt(br.readLine());
                max = Math.max(max, tmp - min);
                min = Math.min(min, tmp);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
 
        System.out.println(max);
    }
}

File: programming-contest/AOJ/ALDS/1_1_D.py
import math
 
n = int(input())
tmp = int(input())
tmpmin = tmp
t1 = int(input())
if t1 < tmp: tmpmin = t1
ans = t1 - tmp
 
 
for i in range(2,n):
    tmp = int(input())
  
    if tmp - tmpmin > ans: ans = tmp - tmpmin
    if tmp < tmpmin: tmpmin = tmp
     
print(int(ans))  
